test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -4461,4 +4471,52 @@ public void testTernaryExpression() {
     test(
         srcs("var a = p ? x : {b: 1}; a.c = 2;"), expected("var a = p ? x : {b: 1}; var a$c = 2;"));
   }
+
+  @Test
+  public void testModuleDynamicImport() {
```

Comment:
Sorry to harp on this so much, but would it be possible to move this test to a separate `Test.java` file that sets `getProcessor` to explicitly add all the extra passes that need to run, a little like the `InlineAndCollapsePropertiesTest` file that runs both `AggressiveInlineAliases` and `CollapseProperties`?

If that were the case, I think we could move most of the code for running these passes out of `CompilerTestCase.java` and just into that one new test file...

Thanks for all the fixes, Chad!  Beyond that possibility, I don't have any further comments for this PR.

Reply:
I split them out - and its way cleaner. Take a look now.